core/src/main/java/org/apache/carbondata/core/index/Segment.java

```diff
@@ -417,4 +417,8 @@ public void setSegmentMetaDataInfo(SegmentMetaDataInfo segmentMetaDataInfo) {
   public boolean isExternalSegment() {
     return isExternalSegment;
   }
+
+  public void setIsExternalSegment(boolean isExternalSegment) {
```

Comment:
check, if you can add Path in LoadMetadetail for external partition also

Reply:
added path in loadMetadetail.